napari_cellseg3d/interface.py

```diff
@@ -47,8 +66,241 @@
 dark_red = "#72071d"  # crimson red
 default_cyan = "#8dd3c7"  # turquoise cyan (default matplotlib line color under dark background context)
 napari_grey = "#262930"  # napari background color (grey)
+napari_param_grey = "#414851"  # napari parameters menu color (lighter gray)
+napari_param_darkgrey = "#202228"  # napari default LineEdit color
 ###############
 
+logger = utils.LOGGER
+
+##################
+# Singleton UI widgets
+##################
+
+
+class QWidgetSingleton(type(QObject)):
+    """
+    To be used as a metaclass when making a singleton QWidget,
+     meaning only one instance exists at a time.
+     Avoids unnecessary memory overhead and keeps user parameters even when a widget is closed
+    """
+
+    _instances = {}
+
+    def __call__(cls, *args, **kwargs):
+        """
+        Ensure only one instance of a QWidget with QWidgetSingleton as a metaclass exists at a time
+
+        """
+        if cls not in cls._instances:
+            cls._instances[cls] = super(QWidgetSingleton, cls).__call__(
+                *args, **kwargs
+            )
+        return cls._instances[cls]
+
+
+##################
+# Screen size adjustment error handler
+##################
+
+
+def handle_adjust_errors(widget, type, context, msg: str):
+    """Qt message handler that attempts to react to errors when setting the window size
+    and resizes the main window"""
+    pass
+    # head = msg.split(": ")[0]
+    # if type == QtWarningMsg and head == "QWindowsWindow::setGeometry":
+    #     logger.warning(
+    #         f"Qt resize error : {msg}\nhas been handled by attempting to resize the window"
+    #     )
+    #     try:
+    #         if widget.parent() is not None:
+    #             state = int(widget.parent().parent().windowState())
+    #             if state == 0:  # normal state
+    #                 widget.parent().parent().adjustSize()
+    #                 logger.debug("Non-max. size adjust attempt")
+    #                 logger.debug(f"{widget.parent().parent()}")
+    #             elif state == 2:  # maximized state
+    #                 widget.parent().parent().showNormal()
+    #                 widget.parent().parent().showMaximized()
+    #                 logger.debug("Maximized size adjust attempt")
+    #     except RuntimeError:
+    #         pass
```

Comment:
This code wasn't functional?

Reply:
It's not broken per se, but I've had two issues with it :
- Inconsistency (it sometimes fails to do its job, maybe I missed a window status case, though usually it's when napari is maximized)
- Causes ImportError when using PyQt6 as backend (this was the reason for commenting, it hindered testing of the plugin on other computers)

I could spend some time fixing it, but those resize errors seem to mainly occur on 4K screens (PyQt5's support of it is not perfect) and less so on more typical resolutions. I can check again on a 1080 screen